python/ray/serve/_private/replica.py

```diff
@@ -812,145 +809,203 @@ async def call_reconfigure(self, user_config: Any):
-                else:
-                    yield result
-        elif inspect.isasyncgen(result_generator):
-            async for result in result_generator:
-                if request_metadata.is_grpc_request:
-                    yield request_metadata.grpc_context, result.SerializeToString()
-                else:
-                    yield result
-        else:
-            raise TypeError(
-                "When using `stream=True`, the called method must be a generator "
-                f"function, but '{user_method.__name__}' is not."
+        result = None
+        asgi_args = None
+        user_method = None
+        receive_task = None
+        user_method_name = "unknown"
+        try:
+            is_asgi_app = isinstance(self._callable, ASGIAppReplicaWrapper)
+            user_method = self._get_user_callable_method(request_metadata.call_method)
+            user_method_name = user_method.__name__
+            user_method_params = inspect.signature(user_method).parameters
+            if request_metadata.is_http_request:
+                assert len(request_args) == 1 and isinstance(
+                    request_args[0], StreamingHTTPRequest
+                )
+                (
+                    request_args,
+                    asgi_args,
+                    receive_task,
+                ) = self._prepare_args_for_http_request(
+                    request_args[0],
+                    request_metadata,
+                    user_method_params,
+                    is_asgi_app=is_asgi_app,
+                    generator_result_callback=generator_result_callback,
+                )
+            elif request_metadata.is_grpc_request:
+                # Ensure the request args are a single gRPCRequest object.
+                assert len(request_args) == 1 and isinstance(
+                    request_args[0], gRPCRequest
+                )
+                request_args, request_kwargs = self._prepare_args_for_grpc_request(
+                    request_args[0], request_metadata, user_method_params
+                )
+
+            result = await self._handle_user_method_result(
+                await self._call_func_or_gen(
+                    user_method, *request_args, **request_kwargs
+                ),
+                user_method_name,
+                request_metadata,
+                generator_result_callback=generator_result_callback,
+                is_asgi_app=is_asgi_app,
+                asgi_args=asgi_args,
             )
 
+        except Exception as e:
+            e = wrap_to_ray_error(user_method_name, e)
+            if request_metadata.is_http_request and asgi_args is not None:
+                result = starlette.responses.Response(
+                    f"Unexpected error, traceback: {e}.", status_code=500
+                )
+                await self._send_user_result_over_asgi(result, asgi_args)
+
+            raise e from None
```

Comment:
Why don't we send anything back if the request is a `gRPC` request?

Reply:
That's a good catch, we should propagate the status/message appropriately

Reply:
this is caught on the proxy side and logged/returned as internal error appropriately

none of this behavior is changed in this pr btw